packit_service/service/api/webhooks.py

```diff
@@ -206,28 +208,94 @@ def post(self):
 
         return "Webhook accepted. We thank you, Gitlab.", HTTPStatus.ACCEPTED
 
-    @staticmethod
-    def validate_token():
+    def create_confidential_issue_with_token(self):
+        project_data = request.json["project"]
+
+        namespace, repo_name = project_data["path_with_namespace"].split("/")
+        http_url = project_data["git_http_url"]
+
+        project_token = ProjectTokenModel.get_project(
+            namespace=namespace, repo_name=repo_name, project_url=http_url
+        )
+
+        if not project_token:
+            token = jwt.encode(
+                {"namespace": namespace, "repo_name": repo_name},
+                config.gitlab_token_secret,
+                algorithm="HS256",
+            )
+
+            ProjectTokenModel.create(
+                token=token,
+                namespace=namespace,
+                repo_name=repo_name,
+                project_url=http_url,
+            )
```

Comment:
Wait, do we need to save it in the database?

- Looks like you are not reading it from the database when we get it.
- With the configured secret token, we can regenerate the token if needed.

That's really nice!

Reply:
One problem with that is that we can potentially create a lot of issues. That can be easily fixed later.

Reply:
Do we really want to do this? While testing, I did exactly this and it created 7 issues in few seconds. On each event, it created an issue.

Reply:
We might skip the database and check if the repo issues if we have created one or not (But this has a bit delay and will have issues with the race conditions).

Reply:
> Do we really want to do this? While testing, I did exactly this and it created 7 issues in few seconds. On each event, it will create an issue.

7 issues? How come we got 7 messages at once?

Reply:
> We might skip the database and check with the repo issues if we have created one or not (But this has a bit delay and will have issues with the race conditions).

We can at least try. The problem is that we can't list confidential issues if we don't have admin permissions.

Reply:
Comment events, the issue created by the Packit itself counted as an event. 😅

Reply:
> What about removing the model for now and fix the duplicate issues later?

Yes okay!